### ISSUE TITLE
Lenovo ThinkPad t460s Samsung SSD locked

### ISSUE THREAD

**Issue by KRegan82** [Drive-Trust-Alliance/sedutil#478]
[revertlog.txt](https://github.com/user-attachments/files/16752845/revertlog.txt)
I have a Samsung m.2 sata ssd that I pulled from my lenovo T460s. The query did not look like this is a compatible drive, but I wanted to file a ticket and include a revertlog anyway.

C:\Windows\system32>sedutil-cli --scan

Scanning for Opal compliant disks
\\.\PhysicalDrive0 No  CT500P1SSD8      R021
\\.\PhysicalDrive1 No  SAMSUNG MZNLN256HCHP-000L7               EMT23L6Q
No more disks present ending scan

C:\Windows\system32>sedutil-cli query \\.\PhysicalDrive1
Argument 1 (query) should be a command

C:\Windows\system32>sedutil-cli --query \\.\PhysicalDrive1
Invalid or unsupported disk \\.\PhysicalDrive1

C:\Windows\system32>sedutil-cli --query \\.\PhysicalDrive1
Invalid or unsupported disk \\.\PhysicalDrive1

C:\Windows\system32>sedutil-cli --yesIreallywanttoERASEALLmydatausingthePSID X1JXCMUFD3MVUKHMZXN585KKZT85L46B \\.\PhysicalDrive1
Invalid or unsupported disk \\.\PhysicalDrive1

C:\Windows\system32>sedutil-cli -vvvvv --yesIreallywanttoERASEALLmydatausingthePSID X1JXCMUFD3MVUKHMZXN585KKZT85L46B \\.\PhysicalDrive1 > revertlog.txt 2>&1

I triple-checked that the PSID was correct. It wouldn't matter though as the disk is unsupported by OPAL.

Either way I wanted to file a ticket to see if it would help anyone who thinks they can get this thing compatible with OPAL. 
Including a picture of the drive details. There is nothing precious to me on this drive. It came with this laptop when I bought it YEARS ago. I replaced it immediately and no issues, but my guess is this laptop was either not sourced 100% legit, or the admin forgot to remove the hardware lock before passing it on. This is more of a "let's see if I can ever get this thing fixed" project, so no sweat either way. Appreciate what you guys do.

![20240826_151819](https://github.com/user-attachments/assets/e5c8bee0-515c-407d-99ec-ea60289ff251)

**Comment by KRegan82** [comment]
I wanted to check back to see if the developers agree that this drive is just not supported by OPAL and that is the issue?

**Comment by scottcmarks** [comment]
Hmm.  With the Opal tag and the PSID, I'm thinking it's a Pyrite drive.  Have you tried running it vs the `develop` branch, which at least recognizes Pyrite drives?  I suggest as a next step that you try running `sedutil-cli -vvvvv --query` using a recent build and post the results of that.

**Comment by KRegan82** [comment]
> Hmm. With the Opal tag and the PSID, I'm thinking it's a Pyrite drive. Have you tried running it vs the `develop` branch, which at least recognizes Pyrite drives? I suggest as a next step that you try running `sedutil-cli -vvvvv --query` using a recent build and post the results of that.

So, I downloaded most recent version for windows and received this output:

Log level set to DBG4
sedutil version : 1.20.0
Creating  DtaResponse()
Creating  DtaResponse()
Creating DtaDevOS::DtaDevOS() \\.\PhysicalDrive1
Creating DtaDiskUSB::DtaDiskUSB() \\.\PhysicalDrive1
Entering DtaDevOS::identify()
Exiting DtaDevOS::identify()
Entering DtaDiskUSB::identify()
Entering DtaDiskUSB::sendCmd
Entering DtaDev::discovery0()
Entering DtaDevOS::sendCmd
Exiting DtaDevOS::sendCmd
Entering DtaDiskUSB::sendCmd
Send D0 request to device failed 2
Entering DtaDev::isPresent() 1
Entering DtaDev::isAnySSC 0
Invalid or unsupported disk \\.\PhysicalDrive1
Destroying DtaDevOS
Destroying DtaDiskUSB
Destroying DtaResponse
Destroying DtaResponse

**Comment by KRegan82** [comment]
> Nevermind my previous comment, I didn't look closely at the picture. Did you try Samsung Magician with this drive?

Not familiar with Samsung magician. I installed it, but what should I be trying there?

I don't think it recognizes the drive at all. Maybe because I am now connecting it via a usb enclosure?

**Comment by youk** [comment]
> Not familiar with Samsung magician. I installed it, but what should I be trying there?

Anything. PSID Revert, Secure Erase..

> Maybe because I am now connecting it via a usb enclosure?

Many USB enclosures won't work properly. Read https://github.com/Drive-Trust-Alliance/sedutil/issues/115 for example.

**Comment by Blacklands** [comment]
Yeah, if you're trying to use an OPAL NVME drive over USB, you either need to switch to Thunderbolt/USB4, or get an enclosure with just the right NVME-USB bridge inside that properly transmits the OPAL commands to the drive. Which is an exercise in itself to find; enclosure manufacturers don't exactly advertise OPAL compatibility (by that I mean, they don't do it at all). Many don't even specify what bridge chip is inside their enclosure in its specs.

There's been a couple GitHub issues here about this over the years, I don't know if anyone ever compiled a list of "confirmed works" bridge chips and/or enclosures using them.
